### DISCUSSION TITLE
Should k8s version be a semver (extracted from cloud provider specific version)?

### DISCUSSION THREAD

**Discussion by pmalek** [Kong/kubernetes-telemetry#18] [General]
Kubernetes returns a version from `/version` API endpoint in using the following schema:

- Example from GKE:

    ```json
    {
      "major": "1",
      "minor": "24",
      "gitVersion": "v1.24.1-gke.1400",
      "gitCommit": "206efe6f8106824435f9e408af6f49e61b30ae54",
      "gitTreeState": "clean",
      "buildDate": "2022-06-13T19:52:07Z",
      "goVersion": "go1.18.2b7",
      "compiler": "gc",
      "platform": "linux/amd64"
    }
    ```

- Example from AWS EKS:

    ```json
    {
      "major": "1",
      "minor": "22+",
      "gitVersion": "v1.22.10-eks-84b4fe6",
      "gitCommit": "cc6a1b4915a99f49f5510ef0667f94b9ca832a8a",
      "gitTreeState": "clean",
      "buildDate": "2022-06-09T18:24:04Z",
      "goVersion": "go1.16.15",
      "compiler": "gc",
      "platform": "linux/amd64"
    }
    ```

Now the question is whether we should return what's verbatim in the `gitVersion` field or should we strip it to contain a semver?

Perhaps we should produce fields from cluster version provider:

- one with whatever was returned in `gitVersion`
- and the second one with a deduced semver

WDYT @Kong/team-k8s ?

**Comment by shaneutt** [comment]
I am in favor of the approach of parsing it for two fields, one with the original full content and one with a parsed semver version. This is more convenient for certain kinds of queries where you care about the Kubernetes version in use without having to write some logic locally to do the parsing.

**Comment by pmalek** [comment]
Created an issue for adding k8s semver in cluster version provider: #19
